modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ParametersResolver.java

```diff
@@ -241,7 +242,16 @@ private ResolverSet getResolverSet(Optional<ParameterizedModel> model, List<Para
     });
 
     checkParameterGroupExclusiveness(model, groups,
-                                     parameters.keySet().stream().map(k -> aliasedParameterNames.getOrDefault(k, k))
+                                     parameters.entrySet().stream().flatMap(entry -> {
```

Comment:
clear as water!! 🌝

Reply:
Good luck to the poor soul that might face this code in the future.